stumpy/motifs.py

```diff
@@ -0,0 +1,394 @@
+
+    atol : float, default None
+        Absolute tolerance (see equation in description).
+        If `None`, defaults to `0.25 * np.std(P)`
+
+    rtol : float, default 1.0
+        Relative tolerance (see equation in description).
+
+    normalize : bool, default True
+        When set to `True`, this z-normalizes subsequences prior to computing distances.
+        Otherwise, this function gets re-routed to its complementary non-normalized
+        equivalent set in the `@core.non_normalized` function decorator.
+
+    Return
+    ------
+    top_motif_indices : ndarray
+        Array of top motif indices. List of the indices of all occurrences of the top
+        k motifs, sorted by distance from the motif representative. The motif
+        representative starts at the first index and is the subsequence with the lowest
+        matrix profile value that is part of a motif.
+
+        There are at most `k` rows, each of which represents
+        one motif. All found occurrences within a distance of `atol + MP * rtol` (where
+        MP is the matrix profile value of the motif representative) are returned. The
+        array has either `max_neighbors` columns if specified, or as many columns as
+        number of occurrences for the motif with most found occurrences. Everything else
+        is filled up with -1.
+
+    top_motif_values : ndarray
+        For every occurrence its distance to the motif representative.
+    """
+    if k < 1:  # pragma: no cover
+        logger.warn("The number of motifs, `k`, must be greater than or equal to 1")
+        logger.warn("`k` has been set to `1`")
+        k = 1
+
+    if T.ndim != 1:  # pragma: no cover
+        raise ValueError(
+            f"T is {T.ndim}-dimensional and must be 1-dimensional. "
+            "Multidimensional motif discovery is not yet supported."
+        )
+
+    if P.ndim != 1:  # pragma: no cover
+        raise ValueError(
+            f"T is {P.ndim}-dimensional and must be 1-dimensional. "
+            "Multidimensional motif discovery is not yet supported."
+        )
+
+    # core.preprocess performs a copy, so modifiying this T will not
+    # change the function input time series
+    m = T.shape[0] - P.shape[0] + 1
+
+    T, M_T, Σ_T = core.preprocess(T[np.newaxis, :], m)
+
+    P = P.copy().astype(float)
+    P = P[np.newaxis, :]
+
+    if excl_zone is None:  # pragma: no cover
+        excl_zone = int(np.ceil(m / 4))
+    if atol is None:  # pragma: no cover
+        atol = 0.25 * np.std(P)
+    if max_occurrences is None:  # pragma: no cover
+        max_occurrences = np.inf
+
+    result = _motifs(
+        T=T,
+        P=P,
+        k=k,
+        M_T=M_T,
+        Σ_T=Σ_T,
+        excl_zone=excl_zone,
+        min_neighbors=min_neighbors,
+        max_occurrences=max_occurrences,
+        atol=atol,
+        rtol=rtol,
+        normalize=normalize,
+    )
+    result = (
+        _create_array_from_jagged_list(result[0], fill_value=-1, dtype=int),
```

Comment:
rename to `jagged_list_to_array` and I think it's better to say `dtype="int64"` and `dtype="float64"`